sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/Models/TimeSeriesId.cs

```diff
@@ -0,0 +1,157 @@
+        /// <param name="key3">The third key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 3 keys.
+        /// </remarks>
+        public TimeSeriesId(object key1, object key2, object key3)
+        {
+            _key1 = (true, key1);
+            _key2 = (true, key2);
+            _key3 = (true, key3);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 1 string key.
+        /// </summary>
+        /// <param name="stringKey1">The first key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 1 string key.
+        /// </remarks>
+        public TimeSeriesId(string stringKey1)
+        {
+            _key1 = (true, stringKey1);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 2 string keys.
+        /// </summary>
+        /// <param name="stringKey1">The first key that identifies a Time Series instance.</param>
+        /// <param name="timeSeriesIdStringProp2">The second key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 2 string keys.
+        /// </remarks>
+        public TimeSeriesId(string stringKey1, string timeSeriesIdStringProp2)
+        {
+            _key1 = (true, stringKey1);
+            _key2 = (true, timeSeriesIdStringProp2);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 3 string keys.
+        /// </summary>
+        /// <param name="timeSeriesIdStringProp1">The first key that identifies a Time Series instance.</param>
+        /// <param name="timeSeriesIdStringProp2">The second key that identifies a Time Series instance.</param>
+        /// <param name="timeSeriesIdStringProp3">The third key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 3 string keys.
+        /// </remarks>
+        public TimeSeriesId(string timeSeriesIdStringProp1, string timeSeriesIdStringProp2, string timeSeriesIdStringProp3)
+        {
+            _key1 = (true, timeSeriesIdStringProp1);
+            _key2 = (true, timeSeriesIdStringProp2);
+            _key3 = (true, timeSeriesIdStringProp3);
+        }
+
+        /// <summary>
+        /// Builds an array to represent a single Time Series Id in order to use it when making calls against the Time
+        /// Series Insights client library.
+        /// </summary>
+        /// <returns>An array representing a single Time Series Id.</returns>
+        public object[] ToArray()
+        {
+            if (_key3.Item1)
```

Comment:
It occurred to me we could save off the array as a field in the ctor (we know in each ctor how many to put into the array. Thoughts?

Reply:
I like the idea. Actually if we also build the string representation of an Id (right now done by GetId), we don't even need the _keyx fields anymore. Constructors will be similar to this:
        public TimeSeriesId(object key1, object key2, object key3)
        {
            _keys = new object[] { key1, key2, key3 };
            _id = $"{key1},{key2},{key3}";
        }

What do you think?

Reply:
I like it.